tests/meta/demultiplex/test_utils.py

```diff
@@ -331,6 +335,85 @@ def test_is_syncing_complete_false(
     assert not is_directory_synced
 
 
+@pytest.mark.parametrize(
+    "source_files,expected_result",
+    [
+        ([DemultiplexingDirsAndFiles.COPY_COMPLETE], True),
+        ([DemultiplexingDirsAndFiles.OUTPUT_FILE_MANIFEST], False),
+        (
+            [
+                DemultiplexingDirsAndFiles.OUTPUT_FILE_MANIFEST,
+                DemultiplexingDirsAndFiles.COPY_COMPLETE,
+            ],
+            False,
+        ),
+        ([DemultiplexingDirsAndFiles.CUSTOM_OUTPUT_FILE_MANIFEST], False),
+        (
+            [
+                DemultiplexingDirsAndFiles.CUSTOM_OUTPUT_FILE_MANIFEST,
+                DemultiplexingDirsAndFiles.COPY_COMPLETE,
+            ],
+            False,
+        ),
+    ],
+)
+def test_needs_manifest_file(source_files: list[str], expected_result: bool, tmp_path):
+    """Tests if a manifest file is needed given the files present."""
+    # GIVEN a source directory
+    source_directory = Path(tmp_path, "source")
+    Path(tmp_path, "source").mkdir()
+
+    # GIVEN the files present
+    for file in source_files:
+        Path(source_directory, file).touch()
+
+    # WHEN checking if a manifest file is needed
+    result = needs_manifest_file(source_directory)
+
+    # THEN the result should as expected
+    assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    "source_files,expected_result",
+    [([DemultiplexingDirsAndFiles.COPY_COMPLETE], True), ([], False)],
+)
+def test_flow_cell_sync_confirmed(source_files: list[str], expected_result: bool, tmp_path):
+    """Tests the check that a flow cell sync has been confirmed."""
+    # GIVEN a flow cell directory with the given file present
+    for file in source_files:
+        Path(tmp_path, file).touch()
+
+    # WHEN checking if the flow cell sync has been confirmed
+    result = flow_cell_sync_confirmed(tmp_path)
+
+    # THEN the result should match what we expect
+    assert result == expected_result
+
+
+def test_create_manifest_file(tmp_flow_cells_directory_ready_for_demultiplexing_bcl_convert: Path):
+    # GIVEN a flow cell directory with files
+    all_files: list[Path] = []
+    for subdir, _, files in os.walk(tmp_flow_cells_directory_ready_for_demultiplexing_bcl_convert):
+        subdir = Path(subdir).relative_to(
+            tmp_flow_cells_directory_ready_for_demultiplexing_bcl_convert
+        )
+        all_files.extend([Path(subdir, file) for file in files])
```

Comment:
same code as in the function - argues more to extract it into a generic function in cg/utils/files to keep in DRY

Reply:
I don't think usage in tests should justify creating separate functions in the production code.

Reply:
Created a helper function for the tests